src/euphorie/client/browser/webhelpers.py

```diff
@@ -1170,5 +1170,146 @@ def is_new_session(self):
         session = self.traversed_session.session
         return session.children().count() == 0
 
+    @property
+    @memoize
+    def survey_tree_data(self):
+        if self.anonymous:
+            return []
+
+        section = self.get_phase()
+        can_edit = self.can_edit_session
+        is_new_session = self.is_new_session
+        url = self.traversed_session.absolute_url()
+        integrated_action_plan = self.integrated_action_plan
+
+        data = []
+
+        active = section == "preparation"
+        disabled = section == ""
+        data.append(
+            {
+                "active": active,
+                "disabled": disabled,
+                "class": f'{"active" if active else ""}{"disabled" if disabled else ""}',  # noqa: E501
+                "id": "step-1",
+                "name": "preparation",
+                "href": f"{url}/@@start#content",
+                "title": api.portal.translate(
+                    _("label_preparation", default="Preparation")
+                ),
+                "has_tree": False,
+            }
+        )
+
+        if self.use_involve_phase:
+            active = section == "involve"
```

Comment:
Reusing the `active` and `disabled` variables might lead to bugs when setting them is omitted by accident.

Reply:
tnx for the review, but I‌ think I'll keep it as it is currently.
Let's do it in another step.
I'm happy for simplifications of this method anyways!